Support network-wide activation/deactivation in multisite

pfefferle opened the pull request:
The activate and deactivate methods now accept a `$network_wide` parameter and handle flushing rewrite rules for all sites in a multisite network when activated or deactivated network-wide.

<!--- Provide a general summary of your changes in the Title above -->

Fixes #2102 

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Added $network_wide parameter to activate() and deactivate() methods
* Implemented multisite support with site switching and rewrite rule flushing for all sites
* Added proper documentation for the new parameters

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [X] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [X] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [X] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

When activated or deactivated network-wide, the plugin now refreshes rewrite rules across all sites.

</details>

pfefferle committed:
Support network-wide activation/deactivation in multisite

The activate and deactivate methods now accept a $network_wide parameter and handle flushing rewrite rules for all sites in a multisite network when activated or deactivated network-wide.

pfefferle commented:
@obenland is this how it should work?

pfefferle committed:
Merge branch 'trunk' into add/flush-rewrite-rules-on-multisite

pfefferle committed:
Merge branch 'trunk' into add/flush-rewrite-rules-on-multisite

obenland commented on includes/class-activitypub.php:
```diff
@@ -59,24 +59,46 @@ public static function init() {
+				\switch_to_blog( $site );
+				self::flush_rewrite_rules();
+				\restore_current_blog();
+			}
```
We might want to do this as a batched job. Not sure. We should definitely add a `! wp_is_large_network()` to the if clause and skip this on large networks.

pfefferle committed:
Prevent network-wide actions on large multisite networks

Adds a check to skip network-wide operations if the multisite network is considered large, improving performance and avoiding potential issues on large WordPress multisite installations.

matticbot committed:
Add changelog

pfefferle committed:
Merge branch 'trunk' into add/flush-rewrite-rules-on-multisite

pfefferle committed:
Merge branch 'trunk' into add/flush-rewrite-rules-on-multisite
